src/Features/Core/Portable/EditAndContinue/EditAndContinueDocumentAnalysesCache.cs

```diff
@@ -161,8 +172,10 @@ private async Task<ImmutableArray<ActiveStatementLineSpan>> GetLatestUnmappedAct
         return activeStatementSpansBuilder.ToImmutableAndClear();
     }
 
-    private AsyncLazy<DocumentAnalysisResults> GetDocumentAnalysisNoLock(Project baseProject, Document document, ImmutableArray<ActiveStatementLineSpan> activeStatementSpans)
+    private AsyncLazy<DocumentAnalysisResults> GetDocumentAnalysisNoLock(Project oldProject, Project newProject, Document? oldDocument, Document? newDocument, ImmutableArray<ActiveStatementLineSpan> activeStatementSpans)
     {
+        var documentId = oldDocument?.Id ?? newDocument!.Id;
```

Comment:
Should there be some sort of assert that the Project and the Document pairs go with each other?  I think with 2 documents and 2 projects, someone else might mix them up.